### PR TITLE
New spec to get satellite version on satellite client

### PR DESCRIPTION
**Pull request by huali027**
Signed-off-by: Huanhuan Li <huali@redhat.com>

### CODE REVIEW

File: insights/parsers/satellite_version_on_client.py
Comment (xiangce) [inline insights/parsers/satellite_version_on_client.py:33]: Since the name of the parser contains "SatelliteVersion" already, how about renaming this attribute as "version"? and add "major", "minor" just like the SatelliteVersion Combiner?
https://github.com/RedHatInsights/insights-core/blob/62dc4062b8808478aecfe737074dec7c83d50eba/insights/combiners/satellite_version.py#L106-L107

Comment (huali027) [reply insights/parsers/satellite_version_on_client.py]: @xiangce, I updated it, and also combine the parser in the combiner SatelliteVersion. Please help to review it again.

File: insights/combiners/satellite_version.py
Comment (xiangce) [inline insights/combiners/satellite_version.py:58]: I believe you should not put the `SatelliteVersionFromClient` here, since the original `SatelliteVersion` combiner is used to indicate the host is a `"Satellite Server"`. After adding this as optional param, the client of Satellite Server will also be treated as a Sat Server, It will break mand existing rules

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/parsers/satellite_version_on_client.py
Comment (xiangce) [reply insights/parsers/satellite_version_on_client.py]: @huali027 - no you should not combine this parser to the SatelliteVersion.

File: insights/combiners/satellite_version.py
Comment (huali027) [reply insights/combiners/satellite_version.py:58]: @xiangce, yes, right, I forgot some rules used it for satellite server only. I'll revert the related code.

**Comment by bfahr** [comment]
@huali027 @xiangce this PR will need to be modified after #3088 is merged.  I will hold off on my review until those changes have been made.

**Comment by bfahr** [comment]
I've closed #3088, but added #3110 as a pattern to follow for custom datasources.  Please modify this PR to follow that pattern.

**Comment by huali027** [comment]
@xiangce  please help to review

**Comment by huali027** [comment]
We can use the spec "branch_info" to get the version info, I'll close it.

**Pull request closed without merging**
